test/api.js

```diff
@@ -420,6 +419,157 @@ test('GET /', function(t) {
   t.end();
 });
 
+test('DELETE /', function(t) {
+  var email = 'foobator@example.com',
+      emailBody;
+
+  t.test('setup', function(t) {
+    smtp.on("startData", function(connection) {
+      emailBody = '';
+    });
+
+    smtp.on("data", function(connection, chunk) {
+      emailBody += chunk.toString();
+    });
+
+    smtp.on("dataReady", function(connection, callback) {
+      callback(null, "ABC1" + Math.abs(Math.random() * 1000000)); // ABC1 is the queue id to be advertised to the client
+    });
+
+    t.end();
+  });
+
+  t.test('requires an `AuthSession` cookie value', function(t) {
+    t.test('missing cookie', function(t) {
+      request({
+        method: 'DELETE',
+        uri: uri,
+        json: true
+      }, function(err, response, body) {
+        t.equal(response.statusCode, 400);
+        t.equal(body.error, 'Bad Request');
+        t.equal(body.message,
+          'child "cookie" fails because ["cookie" is required]');
+        t.end();
+      });
+    });
+
+    t.test('missing `AuthSession` value', function(t) {
+      request({
+        method: 'DELETE',
+        uri: uri,
+        json: true,
+        headers: {
+          cookie: 'foobar'
+        }
+      }, function(err, response, body) {
+        t.equal(response.statusCode, 400);
+        t.equal(body.error, 'Bad Request');
+        t.equal(body.message, 'Invalid cookie header');
+        t.end();
+      });
+    });
+
+    t.test('cookie with malformed `AuthSession` value', function(t) {
+      request({
+        method: 'DELETE',
+        uri: uri,
+        json: true,
+        headers: {
+          cookie: 'AuthSession=foobar'
+        }
+      }, function(err, response, body) {
+        t.equal(response.statusCode, 400);
+        t.equal(body.error, 'Bad Request');
+        t.equal(
+          body.message,
+          'Malformed AuthSession cookie. Please clear your cookies.');
+        t.end();
+      });
+    });
+
+    t.test('cookie with invalid `AuthSession` value', function(t) {
+      request({
+        method: 'DELETE',
+        uri: uri,
+        json: true,
+        headers: {
+          cookie: 'AuthSession=Zm9vYmF0b3JAZXhhbXBsZS5jb206NTVFMzJDODc6lYf4JMY9PZaz5tGyRNi8wmlrcwc'
+        }
+      }, function(err, response, body) {
+        t.equal(response.statusCode, 401);
+        t.equal(body.error, 'Unauthorized');
+        t.end();
+      });
+    });
+
+    t.end
```

Comment:
This one actually covered the previously missing `return`, which is reason enough for me to want to keep it :)